src/analysis/region_analysis/region_analysis.py

```diff
@@ -0,0 +1,400 @@
+                    datetime=dt, model=model
+                )
+                if not preds_data_path.exists():
+                    # if it's not there
+                    warnings.warn(
+                        f'{preds_data_path.parents[0] / preds_data_path.name} does not exist'
+                    )
+                    continue
+                pred_da = self.load_prediction_data(preds_data_path)
+                # compute the statistics
+                (
+                    datetimes, region_name, predicted_mean_value, true_mean_value
+                ) = self.compute_mean_statistics(
+                    region_da=region_da, true_da=true_da, pred_da=pred_da,
+                    region_lookup=region_lookup, datetime=dt
+                )
+                # store as pandas object and add to
+                dfs.append(pd.DataFrame({
+                    'admin_level_name': [admin_level_name for _ in range(len(datetimes))],
+                    'model': [model for _ in range(len(datetimes))],
+                    'datetime': datetimes,
+                    'region_name': region_name,
+                    'predicted_mean_value': predicted_mean_value,
+                    'true_mean_value': true_mean_value,
+                }))
+
+            if dfs == []:
+                print(f'No matching time data found for {model}')
+                print(f'Contents of {model} dir:')
+                print(f'\t{[f.name for f in (self.models_dir / model).iterdir()]}')
+
+            else:
+                df = pd.concat(dfs).reset_index()
+                df = df.sort_values(by=['datetime'])
+                output_filepath = self.out_dir / model / f'{model}_{admin_level_name}.csv'
+                df.to_csv(output_filepath)
+                print(f'** Written {model} csv to {output_filepath.as_posix()} **')
+                all_model_dfs.append(df)
+
+        if all_model_dfs != []:
+            all_model_df = pd.concat(all_model_dfs).reset_index()
+            all_model_df = (
+                all_model_df
+                .sort_values(by=['datetime'])
+                .drop(columns=['index', 'level_0'])
+            )
+            return all_model_df
+        else:
+            print('No DataFrames Created')
+            return None
+
+    @staticmethod
+    def compute_error_metrics(group_model_performance) -> Tuple[float, float, float]:
+        # drop nans
+        group_model_performance = group_model_performance.dropna(how='any')
+
+        rmse = np.sqrt(mean_squared_error(
+            group_model_performance.true_mean_value,
+            group_model_performance.predicted_mean_value
+        ))
+        mae = mean_absolute_error(
+            group_model_performance.true_mean_value,
+            group_model_performance.predicted_mean_value
+        )
+        r2 = r2_score(
+            group_model_performance.true_mean_value,
+            group_model_performance.predicted_mean_value
+        )
+        return rmse, mae, r2
```

Comment:
Can we save these too?
Since the class knows the structure of the `data_dir`, can it save these results to the `analysis` folder under `<experiment>_<model>.csv`or something

Reply:
calculated for all models!